### ISSUE TITLE
"Unable to find corresponding renderer" when attribute nested inside of emphasis tags

### ISSUE THREAD

**Issue by zmiKaminski** [webuni/commonmark-attributes-extension#6]
When using an attribute on a link (`[link](http://url.com){target="_blank"}`) inside of an emphasis tag (`_` or `*`) you receive an error: 
```
Unable to find corresponding renderer for inline type Webuni\CommonMark\AttributesExtension\InlineAttributes
```

Here is a short script to reproduce the error:

__composer.json__
```
{
    "require": {
        "league/commonmark": "^0.15.6",
        "webuni/commonmark-attributes-extension": "^0.5.0"
    }
}
```

__test.php__
```
<?php

require __DIR__ . '/vendor/autoload.php';

use League\CommonMark\Converter;
use League\CommonMark\DocParser;
use League\CommonMark\Environment;
use League\CommonMark\HtmlRenderer;
use Webuni\CommonMark\AttributesExtension\AttributesExtension;

$environment = Environment::createCommonMarkEnvironment();

// Add attributes extension
$environment->addExtension(new AttributesExtension());

$content = [
    "Click here to ... [register now](https://somelink.com){target=\"_blank\"} ... if you haven’t already!",
    "*Click here to ... [register now](https://somelink.com) ... if you haven’t already!*",
    "_Click here to ... [register now](https://somelink.com) ... if you haven’t already!_",
    "*Click here to ... [register now](https://somelink.com){target=\"_blank\"} ... if you haven’t already!*", // throws error
    "_Click here to ... [register now](https://somelink.com){target=\"_blank\"} ... if you haven’t already!_", // throws error
];

// Convert from Markdown to HTML
foreach ($content as $item) {
    try {
        $converter = new Converter(new DocParser($environment), new HtmlRenderer($environment));
        $result = $converter->convertToHtml($item);
        echo $result . PHP_EOL;   
    } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
    }
}
```
_php -v_
```
PHP 7.0.15-0ubuntu0.16.04.4 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15-0ubuntu0.16.04.4, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans
```

**Comment by defr** [comment]
Note: this is due to CommonMark [EmphasisProcessor](https://github.com/thephpleague/commonmark/blob/master/src/Inline/Processor/EmphasisProcessor.php) ; the processInline up there [explicitely](https://github.com/thephpleague/commonmark/blob/master/src/Inline/Processor/EmphasisProcessor.php#L55) removes every element in the stack between its opener and closer elements.

When using an environment created by createCommonMarkEnvironment() , core elements are registered before any other extensions, and thus EmphasisProcessor runs before AttributesInlineProcessor ; concretly, it means that when AttributesInlineProcessor::processInline is reached, its delimiter is gone.

A simple workaround for now is to create the environment manually, registering the AttributeExtension before CommonMarkCoreExtension.

Something like:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

use League\CommonMark\Converter;
use League\CommonMark\DocParser;
use League\CommonMark\Environment;
use League\CommonMark\HtmlRenderer;
use Webuni\CommonMark\AttributesExtension\AttributesExtension;

$environment = new Environment();
$environment->addExtension(new AttributesExtension());
$environment->addExtension(new \League\CommonMark\Extension\CommonMarkCoreExtension());
$environment->mergeConfig([
  'renderer' => [
    'block_separator' => "\n",
    'inner_separator' => "\n",
    'soft_break'      => "\n",
  ],
  'safe'               => false, // deprecated option
  'html_input'         => Environment::HTML_INPUT_ALLOW,
  'allow_unsafe_links' => true,
]);

$content = [
  "Foo [example](https://example.com){target=\"_blank\"} bar",
  "*Foo [example](https://example.com){target=\"_blank\"} bar*",
];

// Convert from Markdown to HTML
foreach ($content as $item) {
    try {
        $converter = new Converter(new DocParser($environment), new HtmlRenderer($environment));
        $result = $converter->convertToHtml($item);
        echo $result . PHP_EOL;   
    } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
    }
}
```

**Comment by hason** [comment]
Fixed in 1.0.0
